### DISCUSSION TITLE
「隐语开源共建计划 OSCP」 第一期

### DISCUSSION THREAD

**Discussion by Candicepan** [secretflow/secretflow#647] [Announcements]
PS：加码任务来啦～感谢社区小伙伴的热情参与～本期额外加码 5 个社区任务（下方任务列表中，序号加粗的任务）欢迎继续认领呀👏

![OSCP-1](https://mdn.alipayobjects.com/huamei_usjdcg/afts/img/A*ZFUkSbiVAH0AAAAAAAAAAAAADo6HAQ/original)

作为隐私计算开源界的一颗新星，隐语一直致力于推动隐私计算行业的发展，同时也在不断发展和完善中。通过开放、透明和共享的方式，协同行业优秀人才协同合作，推动隐私计算领域蓬勃发展，这是隐语开源的意义。因此，隐语社区非常需要也非常欢迎社区小伙伴们一起参与到社区共建中～

为此，隐语社区发布第一期隐语开源共建计划（SecretFlow Open Source Contribution Plan，简称 SF OSCP），包括任务开发、文档翻译、文档修复等多个社区任务，以提高隐语框架易用性及功能丰富度。众人拾柴火焰高，在此特邀请社区小伙伴们一起参与，让我们一起推进隐私计算技术不断发展！

这是一次项目实践经历，更是一次全方位提升能力的机会，参与其中，你可以：
- 结识来自不同背景和领域的开发者
- 参与生产项目具体开发流程，提升技术水平
- 贡献成果将展示给更多的人，这也有助于推广和普及隐私计算技术

在这个夏天，开启你的开源升级打怪之旅吧～

## 具体流程
1. 报名：若您还没有完成报名，请点击[这里](https://www.wjx.top/vm/tUFjR9r.aspx)完成报名 / If you are an overseas developer, you can complete the event registration by sending an email
   - 问卷报名：点击[这里](https://www.wjx.top/vm/tUFjR9r.aspx)
   - Email registration：Send an email to secretflow-contact@service.alipay.com
     - Title：[OSCP] sign up + GitHub ID
     - Text：Name + GitHub ID + Current school/Company + Major/Technical fields
2. 加群：添加小助手微信：secretflow01，备注【OSCP】，即可加入群聊
3. **任务认领**：你可以在 **[GitHub Project](https://github.com/orgs/secretflow/projects/1/)** 中查看本次活动的所有任务，包括详细的任务要求以及认领进度。若你对该任务感兴趣，请在任务的具体 ISSUE 下按格式回复，当该任务将 assign 给你后即为成功认领～
   - 认领回复格式：【GitHub ID + Give it to me】
   - eg：【Candicepan Give it to me】
4. 任务开发：请在任务认领后，**两周内完成相应的任务开发**，否则该任务将被重新释放。在完成任务过程中，有任何问题都可以在微信群/ GitHub 上以 ISSUE、Discussion 形式提问
5. PR 提交：当认领任务要求的任务开发/文档翻译/文档校验任务全部完成后，以 PR 形式提交要求内容到指定位置
说明：收到 PR 提交后，研发老师请在 2-3 个工作日进行验收并给到 Review 意见。

**请注意**：
1. 一次只能认领/提交一个任务。若想要认领多个任务，可以第一个任务完成并通过审核后，再进行第二个任务的认领与提交。
2. 提交 PR 时，请将 PR 与对应任务 ISSUE 进行关联，同时 PR 标题参考 “【OSCP】 + PR 内容 ” ，如 “【OSCP】 使用 SPU 实现逻辑回归算法基础功能” ，以便快速进入 PR review 阶段。
3. 若有任何问题，可以通过微信群/在本 Discussion 下留言，与我们取得联系。
4. 提交 PR 且 PR 被 merger 即为完成该任务。

## 活动时间
- 报名：2023.6.8 - 2023.7.17
- 任务认领与提交：报名后 - 2023.7.17（完成任务即可提交 PR，无需等到 deadline）
- 最终公示：2023.7.22

## 本期任务
任务认领进度以及任务详情均可在 [GitHub Project](https://github.com/orgs/secretflow/projects/1/) 中查看。
本期共有 24 个任务，分为两个难度：简单与中等，其中中等任务为代码开发任务，简单任务为基于文档进行翻译、验证。

No | 类别 | 难度 | 任务名称
-- | -- | -- | --
01 | SPU/SML | 中等 | [使用 SPU 实现逻辑回归算法基础功能](https://github.com/secretflow/spu/issues/211)
02 | SPU/SML | 中等 | [使用 SPU 实现决策树模型基础功能](https://github.com/secretflow/spu/issues/212)
03 | SPU/SML | 中等 | [使用 SPU 实现主成分分析基础功能](https://github.com/secretflow/spu/issues/213)
04 | SF/docs | 简单 | [验证 SecretFlow “快速开始”目录文档，包括验证文档流程和示例代码](https://github.com/secretflow/secretflow/issues/638)
05 | SF/docs | 简单 | [验证 SecretFlow“组件/预处理”目录和“组件/隐私集合求交”的文档及对应教程，包括验证文档流程和示例代码](https://github.com/secretflow/secretflow/issues/639)
06 | SF/docs | 简单 | [验证 SecretFlow“组件/MPC机器学习”目录的文档，包括验证文档流程和示例代码](https://github.com/secretflow/secretflow/issues/640)
07 | SF/docs | 简单 | [验证 SecretFlow“组件/联邦学习/水平联邦”目录的文档及对应教程，包括验证文档流程和示例代码](https://github.com/secretflow/secretflow/issues/641)
08 | SF/docs | 简单 | [验证 SecretFlow“组件/联邦学习/垂直联邦学习”和“组件/联邦学习/混合联邦”的文档及对应教程，包括验证文档流程和示例代码](https://github.com/secretflow/secretflow/issues/642)
09 | SF/docs | 简单 | [验证 SecretFlow 中数据加载的相关教程，包括验证文档流程和示例代码](https://github.com/secretflow/secretflow/issues/643)
10 | SF/docs | 简单 | [验证 SecretFlow 中 SPU 相关教程，包括验证文档流程和示例代码](https://github.com/secretflow/secretflow/issues/644)
11 | SF/docs | 简单 | [验证 SecretFlow 的其他教程，包括验证文档流程和示例代码](https://github.com/secretflow/secretflow/issues/645)
12 | SCQL/docs | 简单 | [将 SCQL《SCQL Introduction》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/62)
13 | SCQL/docs | 简单 | [将 SCQL《Quickstart》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/63)
14 | SCQL/docs | 简单 | [将 SCQL《CCL Setting Suggestions》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/64)
15 | SCQL/docs | 简单 | [将 SCQL《Deployment》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/65)
16 | SCQL/docs | 简单 | [将 SCQL《SCQL Manual》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/66)
17 | SCQL/docs | 简单 | [将 SCQL《SCQL Operators Specification》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/67)
18 | SCQL/docs | 简单 | [将 SCQL《Global Resource Manager》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/68)
19 | SCQL/docs | 简单 | [将 SCQL《SCQL Implementation Status》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/69)
20 | SCQL/docs | 简单 | [将 SCQL《MySQL Compatibility》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/70)
21 | SCQL/docs | 简单 | [将 SCQL《SCQL API》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/71)
22 | SCQL/docs | 简单 | [将 SCQL《How To Integrate SCQL System》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/72)
23 | SCQL/docs | 简单 | [将 SCQL《SCQL Config Manual》英文文档逐句翻译成中文文档](https://github.com/secretflow/scql/issues/73)
24 | HEU/docs | 简单 | [将 HEU 近期新增中文文档译成英文，具体包括：《算法选择》全文《FAQ》小部分中文内容《开发新算法》小部分中文内容](https://github.com/secretflow/heu/issues/72)
**25** | SF/docs | 简单 | [将 SecretFlow《在 SecretFlow 中使用 DataBuilder 进行 SLModel 学习》中文文档逐句翻译成英文文档](https://github.com/secretflow/secretflow/issues/699)
**26** | SF/docs | 简单 | [将 SecretFlow《在 SecretFlow 中使用自定义 DataBuilder（Tensorflow）》中文文档逐句翻译成英文文档](https://github.com/secretflow/secretflow/issues/700)
**27** | SF/docs | 简单 | [将 SecretFlow《在SecretFlow中使用自定义DataBuilder（Torch）》中文文档逐句翻译成英文文档](https://github.com/secretflow/secretflow/issues/701)
**28** | SF/docs | 简单 | [将 SecretFlow《在 SecretFlow 中加载 csv 数据》中文文档逐句翻译成英文文档](https://github.com/secretflow/secretflow/issues/702)
**29** | SF/docs | 简单 | [将 SecretFlow《在 SecretFlow 中加载 Numpy 数据》中文文档逐句翻译成英文文档](https://github.com/secretflow/secretflow/issues/703)

### 其他参考
📖 Git 操作是参与本次任务共建的必备技能哈～可以通过 [using-git](https://docs.github.com/en/get-started/using-git) 了解 git 的基本操作哈～

## 参与收获

为感谢大家对隐语社区的贡献，完成任务开发者可获得：

- 🎁获得隐语社区限量奖品（虽然奖品不多，但我们真心感谢大家的贡献，也欢迎大家有任何意见可以反馈给社区，非常欢迎来自社区的各种建议👏）：
   - 完成 1 个简单任务，即可获得「隐语社区限定款贴纸」&「隐语社区马克杯」
   - 累计完成 2 个及以上简单任务 or 1 个及以上中等任务，即可获得「隐语社区限定款贴纸」&「隐语社区马克杯」&「隐语社区充电宝」&「其他惊喜周边」(关于惊喜周边，也欢迎留言告诉我们你期望的周边类别🎉
- 🆔成为隐语社区 Contributor，且有机会成为隐语社区 Active Contributor、Committer
- 🌟参与隐语社区之星评比

同时，您将享有以下权益：
- 参与内部技术研讨会，与技术专家一起探讨技术前沿和最佳实践；
- 可作为参与嘉宾或社区分享嘉宾，受邀参加/参与分享隐语社区活动，包括但不限于峰会、沙龙、线上直播等；
- 活动期间，**高质量完成任务且为在校学生的开发者**，将有机会加入隐语团队实习。

期待你的参与～

**Comment by yeshuchen323** [comment]
cool~

**Comment by Candicepan** [comment]
[隐语 OSCP 第一期圆满收官](https://mp.weixin.qq.com/s/MR-oJCjeBFjAFJJEwS39ug)，感谢社区小伙伴们的支持 🎉～
